File: java-course/src/test/java/com/example/selenium/spree/BasePage.java
package com.example.selenium.spree;

import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.support.PageFactory;
import org.testng.Assert;

public class BasePage {
    RemoteWebDriver webDriver;
    String URL;
    String title;

    BasePage(RemoteWebDriver webDriver, String URL, String title,
             boolean navigateTo) {
        this.webDriver = webDriver;
        this.URL = URL;
        this.title = title;

        if(navigateTo) {
            navigate();
        } else {
            assertOnPage();
        }

        PageFactory.initElements(this.webDriver, this);
    }

    protected void navigate() {
        webDriver.navigate().to(URL);
        assertOnPage();
    }

    protected void assertOnPage() {
        Assert.assertEquals(webDriver.getTitle(), title);
    }
}

File: python-course/com/example/selenium/spree/BasePage.py
class BasePage(object):
    def __init__(self, test, url, title, navigate_to):
        self.test = test
        self.url = url
        self.title = title
        if navigate_to:
            self.navigate()

    def navigate(self):
        self.test.web_driver.get(self.url)
        self.assert_on_page()

    def assert_on_page(self):
        self.test.assertEquals(self.test.web_driver.title, self.title)
